cmd/livepeer/starter/starter.go

```
@@ -526,16 +526,21 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 		}
 
 	} else {
-		var keystoreDir string
-		if _, err := os.Stat(*cfg.EthKeystorePath); !os.IsNotExist(err) {
-			keystoreDir, _ = filepath.Split(*cfg.EthKeystorePath)
-		} else {
-			keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
-		}
+		var keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
+		if *cfg.EthKeystorePath != "" {
+			keystoreInfo, err := ParseEthKeystorePath(*cfg.EthKeystorePath)
+			if keystoreInfo.path.(string) != "" {
+				keystoreDir = keystoreInfo.path.(string)
+				if keystoreInfo.address.(string) != "" && err == nil {
+					*cfg.EthAcctAddr = keystoreInfo.address.(string)
+				}
+			} else {
+				keystoreDir = filepath.Join(*cfg.Datadir, "keystore")
+			}
 
-		if keystoreDir == "" {
-			glog.Errorf("Cannot find keystore directory")
-			return
+			if err != nil {
+				glog.Warning(fmt.Errorf(err.Error()+" Defaulting to %s", keystoreDir))
```

I think we should fail fast here instead of just logging the warning. This is the case in which a user gave the incorrect `EthKeystorePath`, so the configuration is definitely incorrect.

---

It will now fail if EthKeystorePath does not match a path or file.